modules/compute/gke-node-pool/main.tf

```diff
@@ -379,6 +393,18 @@ resource "google_container_node_pool" "node_pool" {
       condition     = var.enable_flex_start == true ? (var.spot == false) : true
       error_message = "Both enable_flex_start and spot consumption option cannot be set to true at the same time."
     }
+    precondition {
+      condition     = var.reservation_affinity.consume_reservation_type != "SPECIFIC_RESERVATION" || var.static_node_count <= local.reservation_available_count
+      error_message = "Requested static_node_count (${var.static_node_count}) exceeds the available reservation capacity (${local.reservation_available_count})."
+    }
+    precondition {
+      condition     = var.reservation_affinity.consume_reservation_type != "SPECIFIC_RESERVATION" || var.static_node_count == null || var.static_node_count > 0
+      error_message = "static_node_count must be greater than 0 or null when using specific reservations."
+    }
+    precondition {
+      condition     = local.input_specific_reservations_count == 1 || var.reservation_affinity.consume_reservation_type != "SPECIFIC_RESERVATION"
+      error_message = "Exactly one reservation must be specified in var.reservation_affinity.values when using SPECIFIC_RESERVATION."
```

Comment:
" in var.reservation_affinity.values " can be removed as a user potentially does not understand this variable.